### ISSUE TITLE
[Feature Request] Expand Chat Integration to pick up messages from Twitch Extensions

### ISSUE THREAD

**Issue by Runesun** [crowbartools/Firebot#1238]
**Describe the solution you'd like**
It looks like Extension messages in chat are not recognized by Firebot. I expect this has to do with the non-traditional user in chat. (It has a space in the name and an extension marker.) 

I'd like these chat messages to be recognized by Firebot so that I can react to them in events. Specifically for stream stickers they don't have very good reporting, but I could write to a file myself to react to the chat message and I could slap all of the messages from the stream into a text file. Alternatively integrating with stream stickers could work too, but I expect that is probably more work and we'd want to be able to get access to these extension chat messages from inside FireBot regardless.

**Additional context**
Picture shows Firebot chat not recognizing the message and showing the real chat with the extension marker and the name with a space. 

![image](https://user-images.githubusercontent.com/83992396/124369623-a25b3680-dc3b-11eb-9858-c5502026e13a.png)

**Comment by ebiggz** [comment]
Hi @Runesun! Thanks for making this feature request and sorry it took so long to reply here. I've done some research and determined that this is possible for us to implement so we definitely will look at doing so!

**Comment by ebiggz** [comment]
**For devs:** 

Stream Stickers is using a Twitch Extension ability to "Chat as Extension" which they can do via an endpoint. This allows them to chat without having to worry about connecting to IRC.

Extension chat messages are not forwarded to the IRC websocket though. Instead they come through the PubSub via an undocumented topic `"stream-chat-room-v1"`.:

```js
const chatRoomListener = await pubSubClient.onCustomTopic(streamer.userId, "stream-chat-room-v1", (message) => {
            console.log(message);
}, streamer.userId);
```

**Comment by SReject** [comment]
Resolved with v5.52.0

**Comment by Skarnivorous** [comment]
This appears to not be working still in 5.61.2
